### PR TITLE
Blog - How To Work With CSV Files In Go

### PR DESCRIPTION
**Pull request by msali123**
Done with all checks.

### CODE REVIEW

**Comment by msali123** [comment]
@balapriyac [Preview Link](https://64499452a49cb17434bd0005--ephemeral-narwhal-3ac3b2.netlify.app/blog/golang-csv-files/)

File: blog/_posts/2023-06-06-golang-csv-files.md [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+{"age":30,"gender":"Male","name":"David"}]
+~~~
+
+## Conclusion
+
+Working with CSV files in Go is straightforward thanks to the built-in encoding/csv package and third-party libraries such as goCSV. Whether you need to read, append, or convert CSV files, Go provides the tools you need to get the job done. By using the code examples in this article as a starting point, you can explore the benefits that the CSV file format provides for storing and analyzing data.
+
+Mastering these techniques can greatly enhance your ability to work with CSV files in Go.
+
+![Mastering]({{site.images}}{{page.slug}}/mastering.png)\
```
Comment (balapriyac) [inline blog/_posts/2023-06-06-golang-csv-files.md:563]:
```suggestion

```

File: blog/_posts/2023-06-06-golang-csv-files.md [hunk 2]
```diff
@@ -0,0 +1,565 @@
...
+
+~~~{.bash caption=">_"}
+go run main4.go
+~~~
+
+The output should look like this:
+
+![Json to csv]({{site.images}}{{page.slug}}/jSkrisz.png)
+
+## Converting CSV File Into JSON File
```
Comment (balapriyac) [inline blog/_posts/2023-06-06-golang-csv-files.md:409]:
```suggestion
## Converting a CSV File To JSON
```

File: blog/_posts/2023-06-06-golang-csv-files.md [hunk 3]
```diff
@@ -0,0 +1,565 @@
...
+}
+~~~
+
+Go to the terminal and run this;
+
+~~~{.bash caption=">_"}
+go run main3.go
+~~~
+
+## Converting JSON File Into CSV
```
Comment (balapriyac) [inline blog/_posts/2023-06-06-golang-csv-files.md:308]:
```suggestion
## Converting a JSON File To CSV
```

File: blog/_posts/2023-06-06-golang-csv-files.md [hunk 4]
```diff
@@ -0,0 +1,565 @@
...
+
+In the `main` function, we create a new buffer using the `bytes.Buffer` function. We then create a new CSV writer using the `json2csv.NewCSVWriter` function, which takes a buffer as its argument.
+
+Next, we read the JSON data from a file using the `os.ReadFile` function. We use the `json.Unmarshal` function to unmarshal the JSON data into a slice of maps. The maps represent the rows of the CSV file, and the keys represent the columns.
+
+<div class="notice--info">
+In Go, maps are a built-in data structure that associates data in key-value pairs. A map provides constant-time access to its elements, based on their keys. A slice provides a way to reference a subset of an array or map without getting the whole map.
+
+A slice of maps in Go is a data structure that combines the features of slices and maps. It is a dynamic sequence of maps, where each map represents a collection of key-value pairs. In other words, it is a slice where each element is a map.
+</div>
```
Comment (balapriyac) [inline blog/_posts/2023-06-06-golang-csv-files.md:328]:
```suggestion

<div class="notice--info">
### Maps and Slice of Maps in Golang

In Go, **maps** are a built-in data structure that associates data in key-value pairs. A map provides constant-time access to its elements, based on their keys. A slice provides a way to reference a subset of an array or map without getting the whole map.

A **slice of maps** in Go is a data structure that combines the features of slices and maps. It is a dynamic sequence of maps, where each map represents a collection of key-value pairs. In other words, it is a slice where each element is a map.
</div>
```

File: blog/_posts/2023-06-06-golang-csv-files.md [hunk 5]
```diff
@@ -0,0 +1,565 @@
...
+ - Go
+ - Tabular Data
+ - GoCSV
+---
+
+Tabular data is frequently stored in [CSV files](https://en.m.wikipedia.org/wiki/Comma-separated_values), a popular file format. In data processing and analysis projects, managing CSV files is a common task.
+
+And to work with CSV files, a number of packages are available. In this tutorial, we will explore various techniques for managing CSV files using the Go programming language. We will cover topics such as reading and appending CSV files, converting between CSV and JSON file formats.
+
+You can find the code used in this article on [GitHub](https://github.com/khabdrick/go-csv).
```
Comment (balapriyac) [inline blog/_posts/2023-06-06-golang-csv-files.md:19]:
```suggestion

```

File: blog/_posts/2023-06-06-golang-csv-files.md [hunk 6]
```diff
@@ -0,0 +1,565 @@
...
+
+### Prerequisites
+
+To follow along with this tutorial, be sure you have the following:
+
+1. Basic knowledge of Golang
+2. Golang is set up on your local machine
+
```
Comment (balapriyac) [inline blog/_posts/2023-06-06-golang-csv-files.md:27]:
```suggestion
### Prerequisites

To follow along with this tutorial, be sure you have the following:

1. Basic knowledge of Golang
2. Golang is set up on your local machine

You can find the code used in this article on [GitHub](https://github.com/khabdrick/go-csv).
```

File: blog/_posts/2023-06-06-golang-csv-files.md [hunk 7]
```diff
@@ -0,0 +1,565 @@
...
+~~~{.bash caption=">_"}
+go mod init csv
+go get github.com/gocarina/GoCsv
+~~~
+
+First, we import two packages: `os` for file handling and `github.com/gocarina/gocsv` for CSV marshaling and unmarshaling.
+
+Marshaling is the process of converting Go data structures (such as structs, maps, and slices) to a serialized format, typically JSON, XML, or YAML while unmarshaling is the process of decoding data in serialized format into Go data structure.  
+
```
Comment (balapriyac) [inline blog/_posts/2023-06-06-golang-csv-files.md:161]:
```suggestion
<div class="notice--info">
Marshaling is the process of converting Go data structures (such as structs, maps, and slices) to a serialized format, typically JSON, XML, or YAML while unmarshaling is the process of decoding data in serialized format into Go data structure. 
</div>

```

**Review by balapriyac — CHANGES_REQUESTED** [review]
Thanks, Sameer. Suggested a few minor edits.

**Comment by msali123** [comment]
@balapriyac [CHanges DOne](https://644a20b4b909075380cdc6db--ephemeral-narwhal-3ac3b2.netlify.app/blog/golang-csv-files/)

**Review by jalletto — APPROVED** [review]

**Pull request merged**